frontend/src/components/auth/ResetPassword/ForgotPassword.tsx

```diff
@@ -0,0 +1,129 @@
+import React from "react";
+import {
+  Text,
+  FormControl,
+  FormLabel,
+  Input,
+  Button,
+  useToast,
+} from "@chakra-ui/react";
+import { useMutation, useQuery } from "@apollo/client";
+import { RESET_PASSWORD } from "../../../APIClients/mutations/AuthMutations";
+import NavigationButtons from "../TeacherSignup/NavigationButtons";
+import { ArrowBackOutlineIcon } from "../../../assets/icons";
+import { GET_USERS_BY_ROLE } from "../../../APIClients/queries/UserQueries";
+import { UserResponse } from "../../../APIClients/types/UserClientTypes";
+import { Role } from "../../../types/AuthTypes";
+
+interface ForgotPasswordProps {
+  role: Role;
+  hasError: boolean;
+  email: string;
+  step: number;
```

Comment:
is it necessary to include `step` in the state and props? does it make more sense to just refactor the two different "steps" into separate components? based on single-responsibility principle